design/pipeline-artifacts.md

```diff
@@ -15,9 +15,9 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 ### Schema
 
 ```yaml
-- upload: string | [ string ]  # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to upload
+- upload: string # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to upload
   artifact: string # identifier for this artifact (no spaces allowed), defaults to 'default'
-  prependPath: string # a directory path that will be prepended to all uploaded files
+  prependDirectory: string # a directory path that will be prepended to all uploaded files
```

Comment:
Minor suggestion of calling this `prependDirectory` to be clear. I don't feel too strongly about this, but I do think it aligns better with our in-the-box variable names.

When I read the spec, I saw the examples all ended with a trailing `/` to make it clear that it's a directory. And then I had to read the description to understand what the expectation is, if it doesn't end with a trailing `/`.

So my gut feel is, why not name it more intuitively?

Again, just a suggestion. I dont feel too strongly about this.

Reply:
This one I can take or leave. I've used `path` elsewhere even when it's expected to be a directory. Partly, it's easier to spell 😆 But I don't mind "directory" either, as it's more precise.

Reply:
The main argument in favor of `path` is that it's abstract from the OS. For instance, we can say that only forward slashes are accepted as separators (and do the translation on the user's behalf, under the covers, if they're on Windows). Whereas with `directory`, a Windows user is going to expect to be allowed to use backslashes. Not a super strong argument.

Reply:
I'll let @vtbassmatt and @ericsciple make the final call, since you two have a better view of priorities (e.g. consistency vs. precision) across the whole YAML schema.